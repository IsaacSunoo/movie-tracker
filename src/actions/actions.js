export const getMovies = (movies) => ({
    type: 'GET_MOVIES',
    movies
});

export const toggleFavorite = (id) => ({
  type: 'TOGGLE_FAVORITE',
  id
})

export const loginUser = (userInfo) => ({
  type: 'LOGIN_USER',
  userInfo
})