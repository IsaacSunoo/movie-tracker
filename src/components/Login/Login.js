import React, { Component } from 'react';

class Login extends Component {
    constructor() {
        super();
        this.state = {
            email: '',
            password: ''
        }
    }

  handleChange = (e) => {
    const { name, value } = e.target;
    console.log(name, value);
    this.setState({
      [name]: value
    })
  }

  logIn = (e) => {
    e.preventDefault();
    // const { email, password } = this.state;

  }


  render() {
    const { email, password } = this.state;
    return (
      <div className='login-form'>
          <form>
            <label>User Name: <input type='text' name='email' value={email} onChange={this.handleChange}/></label>
            <label>Password: <input type='password' name='password' value={password} onChange={this.handleChange} /></label>
            <button type='submit' className='login-btn'>Login</button>
          </form>
      </div>
    )
  }
}

export default Login;