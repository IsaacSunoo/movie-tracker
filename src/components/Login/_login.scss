.login-form {
    // position: absolute;
    // top: 30px;
    // right: 30px;

    form {
        display: flex;

        input {
            background-color: $flash;
            border: 0;
            font-size: 12px;
            height: 25px;
            margin-top: 4px;
            text-indent: 6px;
            &:focus {
                outline: 0;
    }
        }

        label {
            color: $flash;
            display: flex;
            flex-direction: column;
            margin-right: 20px;
        }
    }
}

.login-btn {
    -moz-box-shadow:inset 0px 1px 0px 0px #7a8eb9;
	-webkit-box-shadow:inset 0px 1px 0px 0px #7a8eb9;
	box-shadow:inset 0px 1px 0px 0px #7a8eb9;
	background:-webkit-gradient(linear, left top, left bottom, color-stop(0.05, #637aad), color-stop(1, #5972a7));
	background:-moz-linear-gradient(top, #637aad 5%, #5972a7 100%);
	background:-webkit-linear-gradient(top, #637aad 5%, #5972a7 100%);
	background:-o-linear-gradient(top, #637aad 5%, #5972a7 100%);
	background:-ms-linear-gradient(top, #637aad 5%, #5972a7 100%);
	background:linear-gradient(to bottom, #637aad 5%, #5972a7 100%);
	filter:progid:DXImageTransform.Microsoft.gradient(startColorstr='#637aad', endColorstr='#5972a7',GradientType=0);
	background-color:#637aad;
	border:1px solid #314179;
    // display:inline-block;
    height: 30px;
	cursor: pointer;
	color: $flash;
	font-family: Arial;
	font-size: 10px;
    font-weight: bold;
    margin: 19px 20px 0 0;
	padding: 2px 12px;
    text-decoration: none;

    &:hover {
        background:-webkit-gradient(linear, left top, left bottom, color-stop(0.05, #5972a7), color-stop(1, #637aad));
	    background:-moz-linear-gradient(top, #5972a7 5%, #637aad 100%);
	    background:-webkit-linear-gradient(top, #5972a7 5%, #637aad 100%);
	    background:-o-linear-gradient(top, #5972a7 5%, #637aad 100%);
	    background:-ms-linear-gradient(top, #5972a7 5%, #637aad 100%);
	    background:linear-gradient(to bottom, #5972a7 5%, #637aad 100%);
	    filter:progid:DXImageTransform.Microsoft.gradient(startColorstr='#5972a7', endColorstr='#637aad',GradientType=0);
	    background-color:#5972a7;
    }

    &:focus {
        outline: 0;
    }

    &:active {
        position:relative;
        top:1px;
    }
}