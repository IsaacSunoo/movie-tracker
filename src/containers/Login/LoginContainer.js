// import { connect } from 'react-redux';
// import { loginUser } from '../../actions/actions';

// export const mapDispatchToProps = (dispatch) => ({
//     loginUser: (userInfo) => dispatch(loginUser(userInfo))
// })

// export default connect(null, mapDispatchToProps)(Login);