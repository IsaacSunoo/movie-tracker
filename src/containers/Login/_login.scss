.login-form {
    form {
        display: flex;

        input {
            background-color: #a5a9d3;
            border: 0;
            font-size: 12px;
            height: 25px;
            margin-top: 4px;
            text-indent: 6px;
            &:focus {
                outline: 0;
            }
        }

        label {
            color: $cream;
            display: flex;
            flex-direction: column;
            margin-right: 20px;
        }
    }
}

.pointer {
    cursor: pointer;
    color: $cream;
}

.login-btn {
    @include btn-design();
}