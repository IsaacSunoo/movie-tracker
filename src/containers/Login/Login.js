import React, { Component } from 'react';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';
import { postFetch } from '../../api';
import PropTypes from 'prop-types';

export class Login extends Component {
         constructor() {
           super();
           this.state = {
             id: 0,
             name: '',
             email: '',
             password: '',
             error: ''
           };
         }

         handleChange = e => {
           const { name, value } = e.target;
           this.setState({
             [name]: value
           });
         };

         handleLogin = async (url, body) => {
           try {
             const response = await postFetch(url, 'POST', body);
             console.log(response);
             if (response) {
               this.setState({ name: response.name, id: response.id });
               const favoritesArray = await this.getFavoriteMovies(this.state.id);
               this.addFavoritesToStore(favoritesArray);
               this.props.loginUser(response);
               this.props.switchDisplay('loggedIn');
             }
           } catch (err) {
             throw new Error(err);
           }
         };

         getFavoriteMovies = async id => {
           const backendUrl = `http://localhost:3000/api/users/${id}/favorites`;
           const options = {
             method: 'GET',
             headers: { 'Content-Type': 'application/json' }
           };
           try {
             const response = await fetch(backendUrl, options);
             const favorites = await response.json();
             return favorites.data;
           } catch (err) {
             console.log(err);
           }
         };

         addFavoritesToStore = (favoritesArray) => {
           favoritesArray.map(movie => {
             return this.props.favorites.push(movie.movie_id)
           })
         }

         logIn = e => {
           e.preventDefault();
           const { email, password } = this.state;
           const url = 'users';
           this.handleLogin(url, { email, password });
         };

         render() {
           const { email, password } = this.state;
           const { switchDisplay } = this.props;
           return (
             <div className="login-form">
               <form onSubmit={this.logIn}>
                 <label>
                   User Email:{' '}
                   <input
                     type="text"
                     name="email"
                     value={email}
                     onChange={this.handleChange}
                   />
                 </label>
                 <label>
                   Password:{' '}
                   <input
                     type="password"
                     name="password"
                     value={password}
                     onChange={this.handleChange}
                   />
                 </label>
                 <button type="submit" className="login-btn">
                   Login
                 </button>
               </form>
               <div
                 onClick={() => {
                   switchDisplay('signup');
                 }}
                 className="pointer"
               >
                 Need an Account?
               </div>
             </div>
           );
         }
       }

export const mapStateToProps = state => ({
  currentUser: state.loginUser,
  favorites: state.favorites
});

export const mapDispatchToProps = (dispatch) => ({
  loginUser: (userInfo) => dispatch(loginUser(userInfo))
});

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  currentUser: PropTypes.object.isRequired
}
export default connect(mapStateToProps, mapDispatchToProps)(Login);