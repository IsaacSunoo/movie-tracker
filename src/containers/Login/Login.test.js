import React from 'react';
import { Login, mapStateToProps, mapDispatchToProps } from './Login';
import { shallow } from 'enzyme';
import * as actions from '../../actions';

let wrapper;
describe('Login', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Login />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      id: 0,
      name: '',
      email: '',
      password: '',
      error: ''
    });
  });

  describe('handleChange', () => {
    it('should update state name according to what is typed', () => {
      const mockEvent = { target: { name: 'name', value: 'Isaac' } };
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('name')).toEqual('Isaac');
    });
    it('should update state password according to what is typed', () => {
      const mockEvent = { target: { name: 'password', value: 'ILoveLamp' } };
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('password')).toEqual('ILoveLamp');
    });
  });

  it('should handleLogin', () => {

  });

  it('should logIn', () => {

  });
  describe('mapStateToProps', () => {
    it('should return an object with currentuser and favorites', () => {
      const mockState = {
        loginUser: {name: 'Isaac'},
        favorites: [123, 444],
        movies: [{title: 'Hey'}, {title: 'anotherMovie'}]
      }
      const expected = {
        currentUser: {name: 'Isaac'},
        favorites: [123, 444]
      }
      const result = mapStateToProps(mockState);
      expect(result).toEqual(expected)
    });
  });
  describe('mapDispatchToProps', () => {
    it('should call dispatch when using loginUser', () => {
      const mockDispatch = jest.fn();
      const action = actions.loginUser({name: 'Mason'})
      const mappedDispatch = mapDispatchToProps(mockDispatch)
      mappedDispatch.loginUser({name: 'Mason'})
      expect(mockDispatch).toHaveBeenCalledWith(action)
    });
  });
});