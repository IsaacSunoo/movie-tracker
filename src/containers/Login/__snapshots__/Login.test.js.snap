// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <label>
            User Email: 
            <input
              name="email"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <label>
            Password: 
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </label>
          <button
            className="login-btn"
            type="submit"
          >
            Login
          </button>
        </form>,
        <div
          className="pointer"
          onClick={[Function]}
        >
          Need an Account?
        </div>,
      ],
      "className": "login-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              User Email: 
              <input
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>,
            <label>
              Password: 
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </label>,
            <button
              className="login-btn"
              type="submit"
            >
              Login
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "User Email: ",
                <input
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "User Email: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password: ",
                <input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Password: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "login-btn",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Need an Account?",
          "className": "pointer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Need an Account?",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <label>
              User Email: 
              <input
                name="email"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
            <label>
              Password: 
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </label>
            <button
              className="login-btn"
              type="submit"
            >
              Login
            </button>
          </form>,
          <div
            className="pointer"
            onClick={[Function]}
          >
            Need an Account?
          </div>,
        ],
        "className": "login-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                User Email: 
                <input
                  name="email"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>,
              <label>
                Password: 
                <input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </label>,
              <button
                className="login-btn"
                type="submit"
              >
                Login
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "User Email: ",
                  <input
                    name="email"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "User Email: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "email",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Password: ",
                  <input
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Password: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "password",
                    "onChange": [Function],
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "login-btn",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Need an Account?",
            "className": "pointer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Need an Account?",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
